lib/pbench/test/unit/common/test_utils.py

```diff
@@ -18,3 +20,47 @@ def test_md5sum():
         assert (
             hash_md5 == expected_hash_md5
         ), f"Expected MD5 '{expected_hash_md5}', got '{hash_md5}'"
+
+
+class TestCleanup:
+    def test_contruct(self, caplog):
+        logger = logging.getLogger("test_construct")
+        c = Cleanup(logger)
+        assert isinstance(c.actions, collections.deque)
+        assert len(c.actions) == 0
+        assert c.logger is logger
```

Comment:
I asked a similar question in Peter's review; I'll ask it here, again:  are these things that we really want to test?  E.g., is it important that an instance of the `Cleanup` class have those attributes with those values?  Are they part of this unit's _interface_?

I suggest that we limit our unit testing to ensuring and enforcing the interface contract for the unit.  Otherwise, we risk making it too hard to revise and refactor the internals of the CUT.

So, I would propose omitting this test (other tests will also ensure that we can construct a `Cleanup` instance...), and try to test the behaviors of the class, rather than making the unit test code a mirror of the CUT.

Reply:
Nevertheless, it **would** be good to have a c'tor test which demonstrates that actions which are not callables are rejected.